x-pack/test/functional/apps/maps/embeddable/embeddable_library.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+
+export default function ({ getPageObjects, getService }) {
+  const find = getService('find');
+  const testSubjects = getService('testSubjects');
+  const PageObjects = getPageObjects(['common', 'dashboard', 'maps']);
+  const kibanaServer = getService('kibanaServer');
+  const security = getService('security');
+  const dashboardAddPanel = getService('dashboardAddPanel');
+  const dashboardPanelActions = getService('dashboardPanelActions');
+
+  describe('maps in embeddable library', () => {
+    before(async () => {
+      await security.testUser.setRoles(
+        [
+          'test_logstash_reader',
+          'global_maps_all',
+          'geoshape_data_reader',
+          'global_dashboard_all',
+          'meta_for_geoshape_data_reader',
+        ],
+        false
+      );
+      await kibanaServer.uiSettings.replace({
+        defaultIndex: 'c698b940-e149-11e8-a35a-370a8516603a',
+      });
+      await PageObjects.common.navigateToApp('dashboard');
+      await PageObjects.dashboard.clickNewDashboard();
+    });
+
+    after(async () => {
+      await security.testUser.restoreDefaults();
+    });
+
+    it('unlink lens panel from embeddable library', async () => {
+      await PageObjects.common.navigateToApp('dashboard');
+      await PageObjects.dashboard.clickNewDashboard();
+      await dashboardAddPanel.clickOpenAddPanel();
+      await dashboardAddPanel.filterEmbeddableNames('document example');
+      await find.clickByButtonText('document example');
+      await dashboardAddPanel.closeAddPanel();
+
+      const originalPanel = await testSubjects.find('embeddablePanelHeading-documentexample');
+      await dashboardPanelActions.unlinkFromLibary(originalPanel);
+      await testSubjects.existOrFail('unlinkPanelSuccess');
```

Comment:
Would it be better instead of asserting a success message gets displayed to also go to visualize listing page and ensure saved object is listed when its in the library and not listed when not in library?

Reply:
Even though unlinking won't actually remove any entries from the library, this is still important to test at least once. You could start by creating a panel by value, adding it via the 'save to library' action, then double checking that it shows up in the visualize listing page.